### PR TITLE
Record info for barriers and mutexes

### PR DESCRIPTION
**Pull request by kravciak**
Local run: http://dhcp165.suse.cz/tests/6056#step/stack_bootstrap/2

### CODE REVIEW

**Comment by coolo** [comment]
perhaps rebase to fix the tests?

**Comment by coolo** [comment]
looks like your test fails

File: lockapi.pm [hunk 1]
```diff
@@ -52,6 +52,25 @@ sub _lock_action {
...
+    my ($name, %args) = @_;
+
+    # Generate log message
+    my $job = $args{where} ? get_job_info($args{where})->{settings}->{TEST} . " #$args{where}" : 'parent job';
+    my $msg = "Wait for $name (on $job)";
+    $msg .= " - $args{info}" if $args{info};
+
+    if (defined $args{amend}) {
+        # amend log info with wait duration
+        $autotest::current_test->remove_last_result;
```
Comment (coolo) [inline lockapi.pm]: Can't call method "record_resultfile" on an undefined value at ../testapi.pm line 213.

Comment (kravciak) [reply lockapi.pm]: Right, I think it is missing $autotest::current_test. I will look at it during the week.
```perl
ok(barrier_wait('barrier1'), 'registered for waiting and released immideately');
  barrier_wait: _log $name, where => $where;
  _log: record_info 'Paused '
  record_info: $autotest::current_test->record_resultfile($title, $output, %nargs);
```

Comment (kravciak) [reply lockapi.pm]: Fixed

**Pull request merged**